Remora.Discord.Commands/Extensions/CommandNodeExtensions.cs

```diff
@@ -42,5 +43,41 @@ public static ApplicationCommandType GetCommandType(this CommandNode node)
             var attribute = node.CommandMethod.GetCustomAttribute<CommandTypeAttribute>();
             return attribute?.Type ?? ApplicationCommandType.ChatInput;
         }
+
+        /// <summary>
+        /// Finds the first custom attribute on the given command or any of its ancestors.
+        /// </summary>
+        /// <typeparam name="T">The type of attribute to search for.</typeparam>
+        /// <param name="node">The command node.</param>
+        /// <param name="includeAncestors">Indicates that ancestors of the command should also be search for the attribute.</param>
+        /// <returns>A custom attribute that matches <typeparamref name="T"/>, or <c>null</c> if no such attribute is found.</returns>
+        public static T? FindCustomAttributeOnLocalTree<T>(this CommandNode node, bool includeAncestors = true) where T : Attribute
+        {
+            // Attempt to first check for ephemerailty on the command itself
```

Comment:
Since this is a generalized method, avoid references to specific implementation details.

Reply:
Ah yes, the good old copy-paste error 😄.